src/main/java/abyss/Abyss.java

```diff
@@ -0,0 +1,94 @@
+            "What can we do for you today?"};
+    private static final String EXIT_MESSAGE = "Exiting the Abyss. We look forward to your return.";
+
+    /**
+     * Main function that handles the flow of the application.
+     *
+     * @param args Input arguments.
+     */
+    public static void main(String[] args) {
+        Ui.printLogo();
+        Ui.reply(START_MESSAGES);
+
+        try {
+            storage = new Storage(FILE_PATH);
+            tasks = storage.loadTasks();
+        } catch (IOException | LoadTaskException e) {
+            Ui.reply(e.getMessage());
+            return;
+        }
+
+        Scanner sc = new Scanner(System.in);
+        while (sc.hasNextLine()) {
+            try {
+                String input = sc.nextLine().trim();
+                Command cmd = Parser.parseCommand(input);
+                Task task;
+
+                if (cmd.getClass() == DoneCommand.class) {
+                    DoneCommand done = (DoneCommand) cmd;
+                    tasks.markAsDone(done.getIndex());
+                    storage.saveTasks(tasks);
+                } else if (cmd.getClass() == DeleteCommand.class) {
+                    DeleteCommand delete = (DeleteCommand) cmd;
+                    tasks.delete(delete.getIndex());
+                    storage.saveTasks(tasks);
+                } else if (cmd.getClass() == TodoCommand.class) {
+                    TodoCommand todo = (TodoCommand) cmd;
+                    task = tasks.addToDo(todo.getDescription());
+                    Ui.replyTaskAdded(task, tasks.getNumberOfTasks());
+                    storage.saveTasks(tasks);
+                } else if (cmd.getClass() == DeadlineCommand.class) {
+                    DeadlineCommand deadline = (DeadlineCommand) cmd;
+                    task = tasks.addDeadline(deadline.getDescription(), deadline.getDate());
+                    Ui.replyTaskAdded(task, tasks.getNumberOfTasks());
+                    storage.saveTasks(tasks);
+                } else if (cmd.getClass() == EventCommand.class) {
+                    EventCommand event = (EventCommand) cmd;
+                    task = tasks.addEvent(event.getDescription(), event.getDate());
+                    Ui.replyTaskAdded(task, tasks.getNumberOfTasks());
+                    storage.saveTasks(tasks);
+                } else if (cmd.getClass() == FindCommand.class) {
+                    FindCommand find = (FindCommand) cmd;
+                    tasks.find(find.getKeyword());
+                }  else if (cmd.getClass() == ListCommand.class) {
+                    tasks.list();
+                } else if (cmd.getClass() == ExitCommand.class) {
+                    Ui.reply(EXIT_MESSAGE);
+                    sc.close();
+                    return;
+                }
+            } catch (AbyssException | IOException | DateTimeParseException e) {
+                Ui.reply(e.getMessage());
+            }
+        }
+    }
+
+    public static int getNumberOfTasks() {
```

Comment:
Do remember to include JavaDoc comments for the remaining methods, once you have the time, of course 😄 💪